core/templates/components/rubrics-editor/rubrics-editor.component.ts

```diff
@@ -95,6 +95,39 @@ export class RubricsEditorComponent {
     return Boolean(this.editableExplanations[difficulty][index]);
   }
 
+  ismediumLevelExplanationLengthEqualsOne(difficulty: string): boolean {
+    if (difficulty === this.skillDifficultyMedium) {
+      return this.rubrics[1]._explanations.length === 1;
+    }
+    return false;
+  }
+
+  isMediumLevelExplanationValid(): boolean {
+    if (this.rubrics[1]._explanations.length >= 1) {
+      return true;
+    }
+    return false;
+  }
+
+  isTotalExplanationsLengthReached(): boolean {
+    if (this.rubrics[0] === undefined || this.rubrics[1] === undefined ||
+      this.rubrics[2] === undefined) {
+      return false;
+    } else {
+      let totalEasyExplanations = this.rubrics[0]._explanations.length;
+      let totalMediumExplanations = this.rubrics[1]._explanations.length;
+      let totalHardExplanations = this.rubrics[2]._explanations.length;
+
+      return (
+        totalEasyExplanations + totalMediumExplanations +
+        totalHardExplanations >= 10);
+    }
+  }
+
+  isExplanationLengthValid(): boolean {
+    return this.editableExplanations[this.rubric.getDifficulty()][0].length <= 300;
+  }
```

Comment:
To keep symmetry with other helper functions, let's also add parameters, `difficulty: string, index: number`. And move this function to be under `isExplanationValid`.